### DISCUSSION TITLE
Send message in channel via a function

### DISCUSSION THREAD

**Discussion by AymenBerb** [discordjs/discord.js#8609] [Q&A] [answered]
I am trying to send a message to a specific channel through a function with parameters.

The connection to the channel does't seem to work inside the function but it works in a client.on block.

I was just wonderign if I am missing something ?

```
// Require the necessary discord.js classes
const { Client, GatewayIntentBits } = require('discord.js');
const { token, clientID } = require('./config.json');

// Create a new client instance
const client = new Client({ 

	intents: [
		GatewayIntentBits.Guilds,
		GatewayIntentBits.GuildMessages,
	],
});

// When the client is ready, run this code (only once)
client.once('ready', () => { 
	console.log('Ready !');
	const ch = client.channels.cache.get(clientID); // connexion works here
	ch.send("Check");
});

client.on('messageCreate', (msg) => {
	// If message sent is from bot, do nothing
	if (msg.author == client.user) {
		return;
	}
	 
	msg.channel.send("Message received, by " + msg.author.toString() + ": " + msg.content); // connexion works here

});

//Method for sending a message to discord channel
function sendMessage() {

	const channel = client.channels.cache.get(clientID); // connexion does not work here
	channel.send("Message");
}

sendMessage();


// Login to Discord with your client's token
client.login(token);
```

**Comment by almostSouji** [comment, marked as answer]
For the same reason as I outlined before:
The channel you are trying to get is not cached at the point this code runs.

Looking at your code, that is further confirmed by it running before the client is even logged in and ready
Which means it runs before it has gotten any information about the guilds (servers) it is in
Which in turn means it runs before there is any channel cached to begin with

Execute that only in a scope that happens after the client is logged in (which is why your comments state that the send does in fact work in those contexts)
